### PR TITLE
SQLite3 snapshot feature added

### PR DESCRIPTION
**Pull request by pintomax**
SQLite3 snapshot feature added

### CODE REVIEW

File: C/plugins/storage/sqlite/common/connection.cpp [hunk 1]
```diff
@@ -2816,3 +2816,113 @@ SQLBuffer	sql;
...
+			 &zErrMsg);
+
+	// Check result code
+	if (rc == SQLITE_OK)
+	{
+		return 1;
+	}
+	else
+	{
+		raiseError("delete_table_snapshot", zErrMsg);
```
Comment (MarkRiddoch) [inline C/plugins/storage/sqlite/common/connection.cpp]: Wrong error source

File: C/plugins/storage/sqlite/common/connection.cpp [hunk 2]
```diff
@@ -2816,3 +2816,113 @@ SQLBuffer	sql;
...
+ * Set the contents of a common table from a snapshot
+ *
+ * @param table		The table to fill
+ * @param id		The snapshot id of the table
+ * @return		-1 on error, >= 0 on success
+ *
+ */
+int Connection::load_table_snapshot(const string& table, const string& id)
+{
+	string query = "INSERT INTO foglamp." + table;
```
Comment (MarkRiddoch) [inline C/plugins/storage/sqlite/common/connection.cpp]: Should delete existing content of table first

File: C/plugins/storage/sqlite/common/connection.cpp [hunk 3]
```diff
@@ -2816,3 +2816,113 @@ SQLBuffer	sql;
...
+			 &zErrMsg);
+
+	// Check result code
+	if (rc == SQLITE_OK)
+	{
+		return 1;
+	}
+	else
+	{
+		raiseError("delete_table_snapshot", zErrMsg);
```
Comment (MarkRiddoch) [inline C/plugins/storage/sqlite/common/connection.cpp]: Wrong error source, should be create_table_snapshot

**Comment by pintomax** [comment]
@amarendra-dianomic Only GET is missing.

It will be added in another task.

This can be merged.

**Pull request merged**